iac-extensions/kubernetes/src/main/java/org/sonar/iac/kubernetes/plugin/HelmProcessor.java

```diff
@@ -90,19 +96,62 @@ private static Map<String, String> validateAndReadFiles(InputFile inputFile, Map
     Map<String, String> fileContents = new HashMap<>(files.size());
 
     for (Map.Entry<String, InputFile> filenameToInputFile : files.entrySet()) {
-      var additionalInputFile = filenameToInputFile.getValue();
-      String fileContent;
-      try {
-        fileContent = additionalInputFile.contents();
-      } catch (IOException e) {
-        throw parseExceptionFor(inputFile, "Failed to read file at " + additionalInputFile, e.getMessage());
+      if (isArchiveFile(filenameToInputFile)) {
+        fileContents.putAll(readCompressedFile(filenameToInputFile.getKey(), filenameToInputFile.getValue()));
+      } else {
+        var fileContent = readTextFile(inputFile, filenameToInputFile.getValue());
+        fileContents.put(filenameToInputFile.getKey(), fileContent);
       }
-
-      fileContents.put(filenameToInputFile.getKey(), fileContent);
     }
     return fileContents;
   }
 
+  private static boolean isArchiveFile(Map.Entry<String, InputFile> filenameToInputFile) {
+    // The files are filter-out before, so it is enough to check only gz suffix, it will match *.tgz and *.gz files
+    return filenameToInputFile.getKey().endsWith("gz");
+  }
+
+  // exposed for tests
+  static Map<String, String> readCompressedFile(String name, InputFile inputFile) {
+    LOG.debug("Read dependency chart {}", name);
+    var pathPrefix = pathPrefix(name);
+    Map<String, String> result = new HashMap<>();
+    try (var source = inputFile.inputStream();
+      var gzip = new GZIPInputStream(source);
+      var tar = new TarArchiveInputStream(gzip)) {
+      TarArchiveEntry entry;
+      while ((entry = tar.getNextEntry()) != null) {
+        var normalizedName = normalizeToRuntimePathSeparator(entry.getName());
+        if (entry.isFile() && HelmFilesystemUtils.includeFile(normalizedName)) {
+          var bytes = tar.readAllBytes();
```

Comment:
There should be protection in how many bytes should be read to avoid high memory consumption.